src/controllers/v1/_common/demand.ts

```diff
@@ -82,15 +86,56 @@ export class DemandController extends Controller {
     if (updated_since) {
       query.push(['updated_at', '>', parseDateParam(updated_since)])
     }
+    const selfAddress = (await this.addressResolver.determineSelfAddress()).address
+    const roles = this.env.ROLES
+    const demands: DemandRow[] = []
+
+    if (roles.includes('optimiser') || roles.includes('admin')) {
+      // Privileged roles get all demands matching base query
+      demands.push(...(await this.db.get('demand', query)))
+      return await Promise.all(demands.map(async (demand) => responseWithAlias(demand, this.identity)))
```

Comment:
can directly map over `await this.db.get('demand', query)`  in the return without pushing to array